In [ ]:
!pip install basemap

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 864 kB 27.0 MB/s 
     |████████████████████████████████| 46 kB 4.3 MB/s 
     |████████████████████████████████| 30.5 MB 1.3 MB/s 
     |████████████████████████████████| 6.3 MB 48.0 MB/s 


In [ ]:
from netCDF4 import Dataset
import numpy as np
import matplotlib.pyplot as plt
from mpl_toolkits.basemap import Basemap
import xarray
import pandas as pd
import os

###Turk_BCC-CSM2 & CESM2

In [ ]:
data = Dataset(r"MOD11A1.061_1km_aid0001.nc")

In [ ]:
data.variables.keys()

dict_keys(['crs', 'time', 'lat', 'lon', 'LST_Day_1km', 'QC_Day'])

In [ ]:
lats=data.variables['lat'][:]
lons=data.variables['lon'][:]
time=data.variables['time'][:]
crs=data.variables['crs'][:]
tmp=data.variables['LST_Day_1km'][:]
data.variables['LST_Day_1km']

<class 'netCDF4._netCDF4.Variable'>
int32 LST_Day_1km(time, lat, lon)
    _FillValue: 0
    coordinates: time lat lon
    grid_mapping: crs
    valid_min: 7500
    valid_max: 65535
    long_name: Daily daytime 1km grid Land-surface Temperature
    units: K
    Number_Type: uint16
    LST: LST data * scale_factor
    scale_factor: 0.02
    scale_factor_err: 0.0
    add_offset_err: 0.0
    calibrated_nt: 5
unlimited dimensions: 
current shape = (365, 363, 362)
filling on

In [ ]:
tmp_c=tmp-273.15

In [ ]:
tmp_c

masked_array(
  data=[[[6.910000000000025, 6.850000000000023, 6.8700000000000045, ...,
          0.9700000000000273, 1.8100000000000023, 2.0300000000000296],
         [6.970000000000027, 6.910000000000025, 6.910000000000025, ...,
          1.1300000000000523, 1.3700000000000045, 1.3700000000000045],
         [6.990000000000009, 6.990000000000009, 6.930000000000007, ...,
          1.0300000000000296, 1.0300000000000296, 1.5500000000000114],
         ...,
         [12.990000000000009, 13.170000000000016, 13.190000000000055,
          ..., 11.270000000000039, 10.870000000000005,
          10.510000000000048],
         [13.470000000000027, 12.990000000000009, 12.990000000000009,
          ..., 12.230000000000018, 11.57000000000005,
          10.870000000000005],
         [12.890000000000043, 12.890000000000043, 12.810000000000002,
          ..., 12.57000000000005, 12.170000000000016,
          11.230000000000018]],

        [[6.830000000000041, 6.850000000000023, 6.670000000000016, ...,
  

In [ ]:
time

masked_array(data=[5479, 5480, 5481, 5482, 5483, 5484, 5485, 5486, 5487,
                   5488, 5489, 5490, 5491, 5492, 5493, 5494, 5495, 5496,
                   5497, 5498, 5499, 5500, 5501, 5502, 5503, 5504, 5505,
                   5506, 5507, 5508, 5509, 5510, 5511, 5512, 5513, 5514,
                   5515, 5516, 5517, 5518, 5519, 5520, 5521, 5522, 5523,
                   5524, 5525, 5526, 5527, 5528, 5529, 5530, 5531, 5532,
                   5533, 5534, 5535, 5536, 5537, 5538, 5539, 5540, 5541,
                   5542, 5543, 5544, 5545, 5546, 5547, 5548, 5549, 5550,
                   5551, 5552, 5553, 5554, 5555, 5556, 5557, 5558, 5559,
                   5560, 5561, 5562, 5563, 5564, 5565, 5566, 5567, 5568,
                   5569, 5570, 5571, 5572, 5573, 5574, 5575, 5576, 5577,
                   5578, 5579, 5580, 5581, 5582, 5583, 5584, 5585, 5586,
                   5587, 5588, 5589, 5590, 5591, 5592, 5593, 5594, 5595,
                   5596, 5597, 5598, 5599, 5600, 56

In [ ]:
mp = Basemap(projection='merc',
            llcrnrlon=-120.00000,
            llcrnrlat=36.00000,
            urcrnrlon=-117.00000,
            urcrnrlat=39.00000,
            resolution = 'i')

In [ ]:
lon,lat=np.meshgrid(lons,lats)
x,y=mp(lon,lat)
plt.figure(figsize=(10,10))

<Figure size 720x720 with 0 Axes>

<Figure size 720x720 with 0 Axes>

In [ ]:
c_scheme=mp.pcolor(x,y,np.squeeze(tmp_c[0,:,:]),cmap='jet')

    
cbar = mp.colorbar(c_scheme,location='right',pad = '10%')
    
meridians= np.arange(-120.14568539999999,-117.03655454062499)
mp.drawmeridians(meridians,labels=[0,0,0,1])
    
parallels = np.arange(36.25770090268727,39.27066973350914)
mp.drawparallels(parallels,labels=[1,0,0,0])

plt.title('Turk_ssp585_CESM2 Near-Surface Air Temperature for Day 1')

plt.clim(-15,15)
    
plt.savefig(r"Heatmap"+'\\'+str(1)+'.jpg')
plt.clf()

<Figure size 432x288 with 0 Axes>

days=np.arange(0,12)

for i in days: 
    c_scheme=mp.pcolor(x,y,np.squeeze(tas[i,:,:]),cmap='jet')

    
    cbar = mp.colorbar(c_scheme,location='right',pad = '10%')
    
    meridians= np.arange(58.582306,61.560469)
    mp.drawmeridians(meridians,labels=[0,0,0,1])
    
    parallels = np.arange(36.25770090268727,39.27066973350914)
    mp.drawparallels(parallels,labels=[1,0,0,0])
    
    day=i+1
    
    if day==1:
        plt.title('Turk_ssp585_CESM2 Near-Surface Air Temperature for Jan')
                  
    elif day==2:
        plt.title('Turk_ssp585_CESM2 Near-Surface Air Temperature for Feb')
        
    elif day==3:
        plt.title('Turk_ssp585_CESM2 Near-Surface Air Temperature for Mar')
    
    elif day==4:
        plt.title('Turk_ssp585_CESM2 Near-Surface Air Temperature for Apr')
    
    elif day==5:
        plt.title('Turk_ssp585_CESM2 Near-Surface Air Temperature for May')
    
    elif day==6:
        plt.title('Turk_ssp585_CESM2 Near-Surface Air Temperature for Jun')
    
    elif day==7:
        plt.title('Turk_ssp585_CESM2 Near-Surface Air Temperature for Jul')
        
    elif day==8:
        plt.title('Turk_ssp585_CESM2 Near-Surface Air Temperature for Aug')
        
    elif day==9:
        plt.title('Turk_ssp585_CESM2 Near-Surface Air Temperature for Sep')
        
    elif day==10:
        plt.title('Turk_ssp585_CESM2 Near-Surface Air Temperature for Oct')
        
    elif day==11:
        plt.title('Turk_ssp585_CESM2 Near-Surface Air Temperature for Nov')
        
    elif day==12:
        plt.title('Turk_ssp585_CESM2 Near-Surface Air Temperature for Dec')
                  
    plt.clim(-20,30)
    
    plt.savefig(r"C:\Users\harry\anaconda3\5900\project data\Turk_ssp585_CESM2"+'\\'+str(day)+'.jpg')
    plt.clf()

c_scheme=mp.pcolor(x,y,np.squeeze(tas[0,:,:]),cmap='jet')
cbar = mp.colorbar(c_scheme,location='right',pad = '10%')
mp.drawcoastlines()
#mp.drawstates()

meridians= np.arange(-120.14568539999999,-117.03655454062499,1)
mp.drawmeridians(meridians,labels=[0,0,0,1])

parallels = np.arange(36.25770090268727,39.27066973350914,1)
mp.drawparallels(parallels,labels=[1,0,0,0])

plt.title('tas on day 1')
plt.show()

days=np.arange(0,365)

for i in days: 
    c_scheme=mp.pcolor(x,y,np.squeeze(tasmin[i,:,:]),cmap='jet')
    mp.drawcoastlines()
    mp.drawstates()
    mp.drawcountries()
    
    cbar = mp.colorbar(c_scheme,location='right',pad = '10%')
    day=i+1
    
    plt.title('Average tasmin for the Day '+str(day))
    plt.clim(-20,20)
    
    plt.savefig(r"C:\Users\harry\anaconda3\5900"+'\\'+str(day)+'.jpg')
    plt.clf()




import PIL

image_frames = [] 
days = np.arange(1,366)

for k in days:
    new_fram = PIL.Image.open(r"C:\Users\harry\anaconda3\5900" + '\\' + str(k) + '.jpg') 
    image_frames.append(new_fram)

image_frames[0].save('tasmin_timelapse.gif',format='GIF',
                    append_images = image_frames[1: ],
                    save_all = True, duration = 100,
                    loop = 0)